Libraries/src/Amazon.Lambda.AspNetCoreServer.Hosting/ServiceCollectionExtensions.cs

```diff
@@ -88,6 +88,73 @@ public static IServiceCollection AddAWSLambdaHosting(this IServiceCollection ser
             return services;
         }
 
+        #if NET8_0_OR_GREATER
+        /// <summary>
+        /// Adds a function meant to initialize the asp.net and lambda pipelines during <see cref="SnapshotRestore.RegisterBeforeSnapshot"/>
+        /// improving the performance gains offered by SnapStart.
+        /// <para />
+        /// Pass a function with one or more <see cref="HttpRequestMessage"/>s that will be used to invoke
+        /// Routes in your lambda function.  The returned <see cref="HttpRequestMessage"/> must have a relative
+        /// <see cref="HttpRequestMessage.RequestUri"/>.
+        /// <para />.
+        /// Be aware that this will invoke your applications function handler code
+        /// multiple times.  Additionally, it uses a mock <see cref="ILambdaContext"/>
+        /// which may not be fully populated.
+        /// <para />
+        /// This method automatically registers with <see cref="SnapshotRestore.RegisterBeforeSnapshot"/>.
+        /// <para />
+        /// If SnapStart is not enabled, then this method is ignored and <paramref name="beforeSnapStartRequests"/> is never invoked.
+        /// <para />
+        /// Example:
+        /// <para />
+        /// <code>
+        /// <![CDATA[
+        /// // Example Minimal Api
+        /// var builder = WebApplication.CreateSlimBuilder(args);
+        ///
+        /// builder.Services.AddAWSLambdaHosting(LambdaEventSource.HttpApi);
+        /// 
+        /// // Initialize asp.net pipeline before Snapshot
+        /// builder.Services.AddAWSLambdaBeforeSnapshotRequest(() => [
+        ///     new HttpRequestMessage(HttpMethod.Get, "/test")
+        /// });
+        /// 
+        /// var app = builder.Build();
+        /// 
+        /// app.MapGet("/test", () => "Success");
+        /// 
+        /// app.Run(); 
+        /// ]]>
+        /// </code>
+        /// </summary>
+        #else
+        /// <summary>
+        /// Snapstart requires your application to target .NET 8 or above.
+        /// </summary>
+        #endif
+        public static IServiceCollection AddAWSLambdaBeforeSnapshotRequest(this IServiceCollection services, Func<IEnumerable<HttpRequestMessage>> beforeSnapStartRequests)
```

Comment:
We should have the method just take in the `HttpRequestMessage` instead of the `Func`. You then don't have a reason to have the overload to take in an `IEnumerable<HttpRequestMessage>` because if a user wants to do multiple requests they can call  `AddAWSLambdaBeforeSnapshotRequest` multiple times.

Reply:
removed overload